packages/wrangler/src/package-manager.ts

```diff
@@ -0,0 +1,105 @@
+import { existsSync } from "node:fs";
+import { join } from "node:path";
+import { execa, execaCommandSync } from "execa";
+
+export interface PackageManager {
+  toString(): string;
+  addDevDeps(...packages: string[]): Promise<void>;
+  install(): Promise<void>;
+}
+
+export async function getPackageManager(root: string): Promise<PackageManager> {
+  const [hasYarn, hasNpm] = await Promise.all([supportsYarn(), supportsNpm()]);
+  const hasYarnLock = existsSync(join(root, "yarn.lock"));
+  const hasNpmLock = existsSync(join(root, "package-lock.json"));
+
+  if (hasYarn && hasYarnLock && !hasNpmLock) {
+    console.log("Using yarn as package manager, as there is a yarn.lock file.");
+    return YarnPackageManager;
+  } else if (hasNpm && hasNpmLock && !hasYarnLock) {
+    console.log(
+      "Using npm as package manager, as there is a package-lock.json file."
```

Comment:
Opinion - we should consider npm as a "default", and not log a message when it's the choice.

Reply:
Funnily enough, when I was playing with this locally, I came to the same conclusions.

Reply:
![image](https://user-images.githubusercontent.com/18808/152352372-efa38a07-e00b-48d8-9abe-a492e9142563.jpeg)